pkg/manifests/manifests.go

```diff
@@ -784,6 +814,187 @@ func (f *Factory) KubeControllersService() (*v1.Service, error) {
 	return f.NewService(MustAssetReader(KubeControllersService))
 }
 
+func (f *Factory) GrafanaClusterRoleBinding() (*rbacv1beta1.ClusterRoleBinding, error) {
+	crb, err := f.NewClusterRoleBinding(MustAssetReader(GrafanaClusterRoleBinding))
+	if err != nil {
+		return nil, err
+	}
+
+	crb.Subjects[0].Namespace = f.namespace
+
+	return crb, nil
+}
+
+func (f *Factory) GrafanaClusterRole() (*rbacv1beta1.ClusterRole, error) {
+	return f.NewClusterRole(MustAssetReader(GrafanaClusterRole))
+}
+
+func (f *Factory) GrafanaConfigConfigMap() (*v1.ConfigMap, error) {
+	c, err := f.NewConfigMap(MustAssetReader(GrafanaConfigConfigMap))
+	if err != nil {
+		return nil, err
+	}
+
+	c.Namespace = f.namespace
+
+	return c, nil
+}
+
+type GrafanaDatasources struct {
+	ApiVersion  int                  `json:"apiVersion"`
+	Datasources []*GrafanaDatasource `json:"datasources"`
+}
+
+type GrafanaDatasource struct {
+	Access            string           `json:"access"`
+	BasicAuth         bool             `json:"basicAuth"`
+	BasicAuthPassword string           `json:"basicAuthPassword"`
+	BasicAuthUser     string           `json:"basicAuthUser"`
+	Editable          bool             `json:"editable"`
+	JsonData          *GrafanaJsonData `json:"jsonData"`
+	Name              string           `json:"name"`
+	OrgId             int              `json:"orgId"`
+	Type              string           `json:"type"`
+	Url               string           `json:"url"`
+	Version           int              `json:"version"`
+}
+
+type GrafanaJsonData struct {
+	TlsSkipVerify bool `json:"tlsSkipVerify"`
+}
+
+func (f *Factory) GrafanaDatasources() (*v1.ConfigMap, error) {
+	c, err := f.NewConfigMap(MustAssetReader(GrafanaDatasourcesConfigMap))
+	if err != nil {
+		return nil, err
+	}
+
+	d := &GrafanaDatasources{}
+	err = json.Unmarshal([]byte(c.Data["prometheus.yaml"]), d)
+	if err != nil {
+		return nil, err
+	}
+	d.Datasources[0].BasicAuthPassword, err = GeneratePassword(43)
```

Comment:
Where does the number 43 originate from?

Reply:
There is no max-length specified in the HTTP basic auth spec. This is up to each individual implementation. Apache seems to do [255](https://stackoverflow.com/questions/4027825/is-there-a-maximum-length-for-a-http-basic-authentication-username), so this seems to be a value worth trying.

What do you think @brancz?

I don't think we are going to hit the [HTTP header size limit](https://stackoverflow.com/questions/686217/maximum-on-http-header-values) here.